mbed_project/mbed_program.py

```diff
@@ -104,9 +120,49 @@ def from_existing_local_program_directory(cls, dir_path: Path) -> "MbedProgram":
         logger.info(f"Found existing Mbed program at path '{program_root}'")
         repo = git.Repo(str(program_root))
         program = MbedProgramData.from_existing(program_root)
-        mbed_os = MbedOS.from_existing(program_root / "mbed-os")
+        mbed_os = MbedOS.from_existing(program_root / MBED_OS_DIR_NAME)
         return cls(repo, program, mbed_os)
 
+    def resolve_libraries(self) -> None:
+        """Clone all external dependencies defined in .lib files.
+
+        This function will recursively clone all unresolved dependencies until all are resolved.
+
+        Raises:
+            VersionControlError: The git clone failed.
+        """
+        for lib in self._find_unresolved_lib_references():
+            git_ref = lib.get_git_reference()
+            print(f"Resolving library reference {git_ref.repo_url}.")
```

Comment:
shouldn't it be a call to the logger?

Reply:
Not sure. I want these messages to appear regardless of log level so I went with `print`.

Reply:
fe70a74